languages/go/internal/ffi/ffi.go

```diff
@@ -131,6 +137,8 @@ func (p PolarFfi) NewQueryFromTerm(queryTerm types.Term) (*QueryFfi, error) {
 	json, err := ffiSerialize(queryTerm)
 	if err != nil {
 		return nil, err
+	} else {
+		defer C.free(unsafe.Pointer(json))
```

Comment:
It's actually fine to call free on a nil pointer. So I would prefer we keep all the frees right next to where we get the string back. Will be easier to scan

Reply:
And avoid the else blocks that are a _little_ hard to follow.